### PR TITLE
Fix missing sounds when interacting with item frames

### PR DESCRIPTION
**Pull request by zSALLAZAR**
### Relevant issues

#5168 

## Tests

https://user-images.githubusercontent.com/59490940/199596126-a545f8cc-c153-4112-a344-55cbea54ea20.mp4

### CODE REVIEW

File: src/network/mcpe/handler/InGamePacketHandler.php [hunk 1]
```diff
@@ -736,7 +737,9 @@ public function handleItemFrameDropItem(ItemFrameDropItemPacket $packet) : bool{
 		$blockPosition = $packet->blockPosition;
 		$block = $this->player->getWorld()->getBlockAt($blockPosition->getX(), $blockPosition->getY(), $blockPosition->getZ());
 		if($block instanceof ItemFrame && $block->getFramedItem() !== null){
-			return $this->player->attackBlock(new Vector3($blockPosition->getX(), $blockPosition->getY(), $blockPosition->getZ()), $block->getFacing());
+			$this->player->getWorld()->addSound($pos = new Vector3($blockPosition->getX(), $blockPosition->getY(), $blockPosition->getZ()), new ItemFrameRemoveItemSound());
```
Comment (JavierLeon9966) [inline src/network/mcpe/handler/InGamePacketHandler.php]: This should be in `ItemFrame->onAttack()`, you don't know if the item frame is empty

**Review by JavierLeon9966 — CHANGES_REQUESTED** [review]

File: src/network/mcpe/handler/InGamePacketHandler.php (hunk 1 above)
Comment (JavierLeon9966) [reply src/network/mcpe/handler/InGamePacketHandler.php]: Now that I see that code above it it does check, but you don't know if a plugin has canceled the interact action

Comment (zSALLAZAR) [reply src/network/mcpe/handler/InGamePacketHandler.php]: You're right
I'll change it tomorrow

File: src/block/ItemFrame.php [hunk 2]
```diff
@@ -162,6 +171,7 @@ public function onAttack(Item $item, int $face, ?Player $player = null) : bool{
 	public function onNearbyBlockChange() : void{
 		if(!$this->getSide(Facing::opposite($this->facing))->isSolid()){
 			$this->position->getWorld()->useBreakOn($this->position);
+			$this->position->getWorld()->addSound($this->position, new ItemFrameRemoveSound());
```
Comment (dktapps) [inline src/block/ItemFrame.php]: `ItemFrameDestroySound` would be more consistent, though I don't see why item frames need a dedicated sound for this. We don't play specific destroy sounds or particles for nearby updates for any other block.

**Comment by dktapps** [comment]
Retargeting this to next-minor since it's adding new APIs.

File: src/block/ItemFrame.php (hunk 2 above)
Comment (zSALLAZAR) [reply src/block/ItemFrame.php]: If you remove the block behind the itemframe in vanilla a itemframe destroy sound is played, but not in Pocketmine, so I added this sound.

**Comment by dktapps** [comment]
I've removed the onNearbyBlockChange changes since those don't directly pertain to interaction, and the missing sounds there are part of a larger issue which should be solved in a more generic way. The rest of the PR is good to merge though.

**Pull request merged**